File: C1.UWP.FlexChart/CS/StockChart/StockChart/Converters/MultipleConverter.cs
﻿using System;
using System.Collections.Generic;
using System.Globalization;
using System.Linq;
using System.Text;
using Windows.UI.Xaml.Data;

namespace StockChart.Converters
{
    class MultipleConverter : IValueConverter
    {
        public object Convert(object value, Type targetType, object parameter, string language)
        {
            double v = System.Convert.ToDouble(value);
            int times = System.Convert.ToInt32(parameter);
            if (times < 1)
            {
                times = 1;
            }
            return v * times;
        }

        public object ConvertBack(object value, Type targetType, object parameter, string language)
        {
            double v = System.Convert.ToDouble(value);
            int times = System.Convert.ToInt32(parameter);
            if (times < 1)
            {
                times = 1;
            }
            return v / times;
        }
    }
}

File: C1.UWP.FlexChart/VB/StockChart/StockChart/Converters/MultipleConverter.vb
Class MultipleConverter
    Implements IValueConverter
    Public Function Convert(value As Object, targetType As Type, parameter As Object, language As String) As Object Implements IValueConverter.Convert
        Dim v As Double = System.Convert.ToDouble(value)
        Dim times As Integer = System.Convert.ToInt32(parameter)
        If times < 1 Then
            times = 1
        End If
        Return v * times
    End Function

    Public Function ConvertBack(value As Object, targetType As Type, parameter As Object, language As String) As Object Implements IValueConverter.ConvertBack
        Dim v As Double = System.Convert.ToDouble(value)
        Dim times As Integer = System.Convert.ToInt32(parameter)
        If times < 1 Then
            times = 1
        End If
        Return v / times
    End Function
End Class